spec/forms/journeys/additional_payments_for_teaching/eligible_itt_subject_form_spec.rb

```diff
@@ -0,0 +1,165 @@
+require "rails_helper"
+
+RSpec.describe Journeys::AdditionalPaymentsForTeaching::EligibleIttSubjectForm, type: :model do
+  before { create(:journey_configuration, :additional_payments) }
+
+  let(:additional_payments_journey) { Journeys::AdditionalPaymentsForTeaching }
+
+  let(:ecp_trainee_teacher_eligibility) do
+    create(
+      :early_career_payments_eligibility,
+      :trainee_teacher
+    )
+  end
+
+  let(:ecp_trainee_teacher_claim) do
+    create(
+      :claim,
+      :first_lup_claim_year,
+      policy: Policies::EarlyCareerPayments,
+      eligibility: ecp_trainee_teacher_eligibility
+    )
+  end
+
+  let(:ecp_qualified_teacher_eligibility) do
+    create(
+      :early_career_payments_eligibility,
+      :eligible_now,
+      :sufficient_teaching
+    )
+  end
+
+  let(:ecp_qualified_teacher_claim) do
+    create(
+      :claim,
+      policy: Policies::EarlyCareerPayments,
+      eligibility: ecp_qualified_teacher_eligibility
+    )
+  end
+
+  let(:current_claim) { CurrentClaim.new(claims: [claim]) }
+
+  describe "validations" do
+    let(:claim) { ecp_qualified_teacher_claim }
+
+    subject(:form) do
+      described_class.new(
+        journey: additional_payments_journey,
+        claim: current_claim,
+        params: ActionController::Parameters.new
+      )
+    end
+
+    it do
+      is_expected.to validate_inclusion_of(:eligible_itt_subject)
+        .in_array(form.available_options)
+        .with_message("Select a subject")
+    end
+  end
+
+  describe "#available_subjects" do
+    let(:form) do
+      described_class.new(
+        journey: additional_payments_journey,
+        claim: current_claim,
+        params: ActionController::Parameters.new
+      )
+    end
+
+    subject(:available_subjects) { form.available_subjects }
+
+    context "when qualified teacher" do
+      let(:claim) { ecp_qualified_teacher_claim }
+
+      # EarlyCareerPayments policy, claim year 2023, itt_year 2020
+      # see `lib/journey_subject_eligibility_checker.rb`
+      it do
+        is_expected.to contain_exactly(
+          "chemistry",
+          "foreign_languages",
+          "mathematics",
+          "physics"
+        )
+      end
+    end
+
+    context "when trainee teacher" do
+      context "when in ECP and LUP policy year range" do
+        let(:claim) { ecp_trainee_teacher_claim }
+
+        it do
+          is_expected.to contain_exactly(
+            "chemistry",
+            "computing",
+            "mathematics",
+            "physics"
+          )
+        end
+      end
+
+      context "when not in ECP and LUP policy year range" do
+        let(:claim) { ecp_trainee_teacher_claim }
+
+        before do
+          allow(current_claim).to(
+            receive(:policy_year).and_return(AcademicYear.new(2000))
+          )
+        end
+
+        it { is_expected.to be_empty }
+      end
+    end
+  end
+
```

Comment:
I think that there are a few more instance methods on the form that we can probably test here